decoderpool.go

```diff
@@ -96,3 +96,32 @@ func NewDecoderPool(options ...zstd.DOption) *sync.Pool {
 
 	return p
 }
+
+// DecoderPoolWrapper is a convenience wrapper for sync.Pool which only
+// accepts and returns *DecoderWrapper's.
+type DecoderPoolWrapper struct {
+	pool *sync.Pool
+}
+
+// Get returns a *DecoderWrapper that has been Reset to use r.
+func (d *DecoderPoolWrapper) Get(r io.Reader) *DecoderWrapper {
+	dw := d.pool.Get().(*DecoderWrapper)
+	_ = dw.Reset(r)
```

Comment:
Hiding this error doesn't feel super great... it looks like the main reason it would return an error is if the underlying decoder was in a permanently broken state. Maybe this function should bubble that error up to the caller? Or maybe it should keep fetching decoders from the pool and attempting to reset them until it finds one that doesn't error? I'm not sure what your thoughts are on this, but hiding the error could lead the caller being given a broken decoder and not knowing it.